src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -64,7 +65,35 @@ dnnl::engine& ocl_engine::get_onednn_engine() const {
         if (!casted)
             throw ov::Exception("[GPU] Invalid device type stored in ocl_engine");
 
-        _onednn_engine = std::make_shared<dnnl::engine>(dnnl::ocl_interop::make_engine(casted->get_device().get(), casted->get_context().get()));
+        auto config = this->configuration();
+        if (config.kernels_cache_path.empty()) {
+            _onednn_engine = std::make_shared<dnnl::engine>(dnnl::ocl_interop::make_engine(casted->get_device().get(), casted->get_context().get()));
+        } else {
+            // Use cached blob
+            auto path = config.kernels_cache_path;
+            if (path.back() != '/' && path.back() != '\\') {
+                path += "/";
+            }
+
+            auto blob_id = dnnl::ocl_interop::get_engine_cache_blob_id(casted->get_device().get());
+            if (blob_id.empty())
+                throw ov::Exception("[GPU] Invalid blob ID");
```

Comment:
nit: From the spec document, it is OK to be empty. I think proper behavior is just to create engine without cache_blob.  https://oneapi-src.github.io/oneDNN/namespace_dnnl_ocl_interop.html#doxid-namespacednnl-1-1ocl-interop-1a969a55a4a7d84dee549ebd2c3bcc2518

Reply:
Fixed